lib/fsm/follow.go

```diff
@@ -59,66 +61,82 @@ func (e *PlanRolledBackEvent) isTerminalEvent() bool { return true }
 
 // FollowOperationPlan returns a channel that will be receiving phase updates
 // for the specified plan.
+// The returned channel must be served to avoid blocking the producer as
+// no events are dropped.
 //
 // The watch will stop upon entering one of the terminal operation states, for
-// example if the obtained plan is completed or fully rolled back.
+// example if the obtained plan is completed or fully rolled back or when the specified
+// context has expired. In any case, the channel will be closed to signal completion
```

Comment:
It's probably the case but I haven't tried to check it, how sure are we that the context always gets cancelled if the client drops?

Reply:
Not sure I follow - drops as in stops servicing the channel from `FollowOperationPlan` without also cancelling the context?
Currently the only client is the `plan --tail` command. Do you mean something else?

Reply:
Sorry, for some reason I thought this was the service side of the call. If this is all running client side then it shouldn't be an issue.